src/com/amazon/ion/impl/SymbolTableStructCache.java

```diff
@@ -0,0 +1,134 @@
+     * @return an IonStruct representing the symbol table.
+     */
+    public IonStruct getIonRepresentation(ValueFactory factory) {
+        synchronized (this) {
+            if (image == null) {
+                makeIonRepresentation(factory);
+            }
+            return image;
+        }
+    }
+
+    /**
+     * @return true if an IonStruct has already been created for the symbol table; otherwise, false.
+     */
+    public boolean hasStruct() {
+        return image != null;
+    }
+
+    /**
+     * Create a new IonStruct representation of the symbol table.
+     * @param factory the {@link ValueFactory} from which to construct the IonStruct.
+     */
+    private void makeIonRepresentation(ValueFactory factory) {
+        image = factory.newEmptyStruct();
+
+        image.addTypeAnnotation(ION_SYMBOL_TABLE);
+
+        if (importedTables.length > 0) {
+            // The system symbol table may be the first import. If it is, skip it.
+            int i = importedTables[0].isSystemTable() ? 1 : 0;
+            if (i < importedTables.length) {
+                IonList importsList = factory.newEmptyList();
+                while (i < importedTables.length) {
+                    SymbolTable importedTable = importedTables[i];
+                    IonStruct importStruct = factory.newEmptyStruct();
+
+                    importStruct.add(NAME,
+                        factory.newString(importedTable.getName()));
+                    importStruct.add(VERSION,
+                        factory.newInt(importedTable.getVersion()));
+                    importStruct.add(MAX_ID,
+                        factory.newInt(importedTable.getMaxId()));
+
+                    importsList.add(importStruct);
+                    i++;
+                }
+                image.add(IMPORTS, importsList);
+            }
+        }
+
+        if (symbolTable.getMaxId() > symbolTable.getImportedMaxId()) {
+            Iterator<String> localSymbolIterator = symbolTable.iterateDeclaredSymbolNames();
+            int sid = symbolTable.getImportedMaxId() + 1;
+            while (localSymbolIterator.hasNext()) {
+                addSymbol(localSymbolIterator.next(), sid);
+                sid++;
+            }
+        }
+    }
+
+    /**
+     * Adds a new symbol table with the given symbol ID to the IonStruct's 'symbols' list.
+     * @param symbolName can be null when there's a gap in the local symbols list.
+     * @param sid the symbol ID to assign to the new symbol.
+     */
+    void addSymbol(String symbolName, int sid) {
+        assert sid >= firstLocalSid;
+
+        ValueFactory sys = image.getSystem();
+
+        IonValue syms = image.get(SYMBOLS);
+        while (syms != null && syms.getType() != IonType.LIST) {
+            image.remove(syms);
+            syms = image.get(SYMBOLS);
+        }
+        if (syms == null) {
+            syms = sys.newEmptyList();
+            image.put(SYMBOLS, syms);
+        }
```

Comment:
Could you add a comment here? It looks like this removes the `symbols` field from the `image` struct repeatedly until it finds one with a list value, creating on if needs be. This seems defensive, but why would `image` be a malformed struct?

Reply:
This is moved from `LocalSymbolTableAsStruct`, where it had some commentary about not handling the case where there were multiple lists. I don't understand the workflow here, so I'd also like to know the answer to Zack's question.

Reply:
I've added the following comment:
```
// If the user has manually created a symbol table via an IonStruct, it may be malformed.
// The following removes any "symbols" fields that aren't lists, then creates a list for
// the symbols field if necessary.
```

The key takeaway is that there is a use case (the "tree" writer - IonWriterSystemTree) that allows users to use the IonWriter interface to create DOM values. Users may create the IonStruct representation of a symbol table that ends up in this method. This defends against that manually-constructed symbol table being malformed.